### PR TITLE
Update error messages to reflect why test is skipped

### PR DESCRIPTION
**Pull request by mikekgfb**
Summary: Update error messages to reflect why test is skipped

Test Plan: github

Differential Revision: D43386390

### CODE REVIEW

File: test/test_transformers.py [hunk 1]
```diff
@@ -1056,7 +1056,7 @@ def ones_tensor(*shape):
                         _ = mha_f(qkv_f, qkv_f, qkv_f, need_weights=False, is_causal=True)
                         torch.cuda.synchronize()
 
-    @unittest.skipIf(not PLATFORM_SUPPORTS_FUSED_SDPA or not SM80OrLater, "CUDA unavailable")
+    @unittest.skipIf(not PLATFORM_SUPPORTS_FUSED_SDPA or not SM80OrLater, "platform does not supposrt SDPA or pre-SM80 hardware")
```
Comment (nmacchioni) [inline test/test_transformers.py:1059]: *support

**Comment by nmacchioni** [comment]
Can we make the messages consistent?

i.e. 
`@unittest.skipIf(not PLATFORM_SUPPORTS_FUSED_SDPA or not SM80OrLater` => `platform does not support fused sdpa or <SM80`,
`@unittest.skipIf(not PLATFORM_SUPPORTS_FUSED_SDPA or not SM86OrLater` => `platform does not support fused sdpa or <SM86`,
`@unittest.skipIf(not PLATFORM_SUPPORTS_FUSED_SDPA` => `platform does not support fused sdpa`,

**Review by nmacchioni — APPROVED** [review]

**Review by cpuhrsch — APPROVED** [review]

**Comment by seemethere** [comment]
Heads up, I'm force reverting this since it appears as though we can't revert using the normal tooling

**Comment by seemethere** [comment]
Reverted here: https://github.com/pytorch/pytorch/commit/e44737e61975c49b20fa176e77c012b626c5f331

**Comment by seemethere** [comment]
@mikekgfb Let's not skip lint failures on these types of PRs in the future. The job should be high signal and should help us avoid issues like this in the future.

**Pull request closed without merging**